uk.co.compucorp.civicrm.hrleaveandabsences/js/angular/src/leave-absences/manager-leave/components/manager-leave-calendar.js

```diff
@@ -354,10 +335,22 @@ define([
           to: vm.selectedPeriod.end_date
         }
       })
-        .then(function (leaveRequestsData) {
-          leaveRequests = leaveRequestsData.list;
-          return loadCalendar();
+      .then(function (leaveRequestsData) {
+        leaveRequests = leaveRequestsData.list;
```

Comment:
There is obviously a great deal of stuff going on in this `then()`. Please move it into a documented function, as `loadLeaveRequestsAndCalender` is becoming too long